lib/plugins/aws/package/compile/events/cognitoUserPool/index.js

```diff
@@ -115,6 +116,12 @@ class AwsCompileCognitoUserPoolEvents {
         DependsOn,
       };
 
+      // If overrides exist in `Resources`, merge them in
+      if (_.has(this.serverless.service.resources, userPoolLogicalId)) {
```

Comment:
The merge of `DependsOn` values present in the overrides is not handled correctly. If the override specifies a `DependsOn: "ResA"` or `DependsOn: [ "ResB", "ResC" ]`, the resulting DependsOn shoud be a concatenation of the of the values in the override and of the resource itself. As far as I know, `_.merge()` does not merge arrays but performs an assign.
You have to merge the dependencies in the override manually here. And, important, DependsOn can either be a string, or an array. The implementation should handle both cases.